app/javascript/components/region-form/index.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import createSchema from './region-form.schema';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import { API } from '../../http_api';
+import miqRedirectBack from "../../helpers/miq-redirect-back";
+
+const RegionForm = ({ id, maxDescLen }) => {
+  const [initialValues, setInitialValues] = useState({});
+  const [isLoading, setIsLoading] = useState(!!id);
+
+  useEffect(() => {
+    if (id) {
+      miqSparkleOn();
+      API.get(`/api/regions/${id}?attributes=description`)
+        .then(({
+                 ...data
+               }) => setInitialValues({
+          ...data,
+        }))
+        .then(() => setIsLoading(false))
+        .then(miqSparkleOff);
+    }
+  }, []);
+
+  const onCancel = () => {
+    const message = __('Edit of Region was cancelled by the user');
+    miqRedirectBack(message, 'success', `/ops/explorer/?button=cancel`);
+  };
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+    const message = __('Region was saved');
+    const submitUrl = `/ops/explorer/${id}?button=save`;
+    const request = () => API.post(`/api/regions/${id}`, {
```

Comment:
No need for the `request` variable allocation here, you can chain like: 
```
return API.post().then().catch()
```

as you're not really using the request afterwards.